e2e/LongHaul/device/IotHub.cs

```diff
@@ -244,6 +218,25 @@ public async Task ReportReadOnlyPropertiesAsync(Logger logger, CancellationToken
             _messagesToSend.Enqueue(iotMessage);
         }
 
+        public async Task ReportReadOnlyPropertiesAsync(Logger logger, CancellationToken ct)
+        {
+            logger.LoggerContext.Add(OperationName, ReportTwinProperties);
+            var sw = new Stopwatch();
+
+            while (!ct.IsCancellationRequested)
+            {
+                sw.Restart();
+                await SetPropertiesAsync("totalTelemetryMessagesSent", _totalTelemetryMessagesSent, logger, ct).ConfigureAwait(false);
```

Comment:
Use the helper method we have for setting properties which will guard against exceptions.